Fix HTML rendering of relationship variables in CiceroMark

Aradhana2803 opened the pull request:
# Fix HTML rendering of relationship variables in CiceroMark

Related to accordproject/template-playground#13 

### Summary
Relationship-typed CiceroMark variables were rendered in HTML as quoted
`resource:` strings instead of a readable identifier. This PR fixes the
HTML rendering logic so relationship variables display correctly.[TemplateMark not rendering relationships "resource" data  #13]


### Changes
- Update `ToHtmlStringVisitor` to detect relationship variables via `elementType`
- Render a clean relationship identifier instead of the raw serialized value
- Preserve existing rendering behavior for primitive variable types

### Flags
- No breaking changes
- Change is limited to HTML rendering only (CiceroMark JSON unchanged)

### Screenshots or Video
Not applicable — behavior verified via existing HtmlTransformer test suite.

### Author Checklist
- [x] Ensure you provide a DCO sign-off for your commits using the `--signoff` option of git commit
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow AP format
- [x] Extend the documentation, if necessary
- [x] Merging to `main` from `fork:branchname`

Commit:
Fix HTML rendering of relationship variables in CiceroMark

Signed-off-by: aradhana2803 <aradhanasingh2803@gmail.com>

git push --force origin fix/relationship-variable-html

Aradhana2803 commented:
This PR fixes the issue reported in template-playground#13 where relationship-typed CiceroMark variables were rendered as quoted resource: strings instead of a readable identifier. HTML rendering now preserves the identifier cleanly, while keeping other variable types unchanged.Verified that `Party` relationship variables now render clean identifiers in HTML
without quotes. All existing HtmlTransformer tests pass.

sanketshevkar reviewed (changes requested):
Can you please add unit tests covering this case?

mttrbrts committed:
fix(markdown-html): use identifiedBy to detect relationship variables

Replace hardcoded namespace prefix check with identifiedBy, which is the
canonical Concerto marker for relationship/reference types. Only strip the
resource URI (resource:namespace#identifier) scheme when the value actually
carries that prefix, leaving primitive quoted values unchanged so existing
tests are unaffected.

Add unit tests covering quoted resource URI, bare resource URI, and
primitive variable rendering.

Signed-off-by: Matt Roberts <matt@rbrts.uk>
Signed-off-by: Matt Roberts <code@rbrts.uk>

mttrbrts commented:
@sanketshevkar I've pushed an updated commit on top of this branch that addresses your review.

**`renderVariableValue` — more robust relationship detection**

The original check used `thing.elementType.startsWith('org.accordproject.party@')`, which only handled one hardcoded namespace. I've replaced it with a check on `thing.identifiedBy`, which is the canonical Concerto marker for relationship/reference types regardless of namespace. The logic now only rewrites the display value when it's a full resource URI (`resource:namespace#identifier`), extracting the identifier after `#`. Plain quoted values like `"Party A"` are passed through unchanged, so existing roundtrip tests are unaffected.

**Unit tests added**

Three cases in a new `describe` block in `HtmlTransformer.test.js`:
1. Quoted resource URI value → renders as clean identifier, no `resource:` prefix
2. Bare (unquoted) resource URI value → same
3. Primitive variable (no `identifiedBy`) → value unchanged

All 65 existing tests still pass.

mttrbrts committed:
fix(markdown-html): add JSDoc to renderVariableValue for require-jsdoc lint rule

Signed-off-by: Matt Roberts <matt@rbrts.uk>
Signed-off-by: Matt Roberts <code@rbrts.uk>
